src/CodeExplorerBundle/Twig/SourceCodeExtension.php

```diff
@@ -0,0 +1,144 @@
+        if (null === $this->controller) {
+            return;
+        }
+
+        $method = $this->getCallableReflector($this->controller);
+
+        $classCode = file($method->getFilename());
+        $methodCode = array_slice($classCode, $method->getStartline() - 1, $method->getEndLine() - $method->getStartline() + 1);
+        $controllerCode = '    '.$method->getDocComment()."\n".implode('', $methodCode);
+
+        return array(
+            'file_path' => $method->getFilename(),
+            'starting_line' => $method->getStartline(),
+            'source_code' => $this->unindentCode($controllerCode)
+        );
+    }
+
+    /**
+     * Gets a reflector for a callable.
+     *
+     * This logic is copied from Symfony\Component\HttpKernel\Controller\ControllerResolver::getArguments
+     *
+     * @param callable $callable
+     *
+     * @return \ReflectionFunctionAbstract
+     */
+    private function getCallableReflector($callable)
+    {
+        if (is_array($callable)) {
+            return new \ReflectionMethod($callable[0], $callable[1]);
+        }
+
+        if (is_object($callable) && !$callable instanceof \Closure) {
+            $r = new \ReflectionObject($callable);
+
+            return $r->getMethod('__invoke');
+        }
+
+        return new \ReflectionFunction($callable);
+    }
+
+    private function getTemplateSource(\Twig_Template $template)
+    {
+        return array(
+            // Twig templates are not always stored in files, and so there is no
+            // API to get the filename from a template name in a generic way.
+            // The logic used here works only for templates stored in app/Resources/views
+            // and referenced via the "filename.html.twig" notation, not via the "::filename.html.twig"
+            // one or stored in bundles. This is enough for the needs of the demo app.
+            'file_path' => $this->kernelRootDir.'/Resources/views/'.$template->getTemplateName(),
+            'starting_line' => 1,
+            'source_code' => $template->getSource(),
+        );
+    }
+
+    /**
+     * Utility method that "unindents" the given $code when all its lines start
+     * with a tabulation of four white spaces.
+     *
+     * @param  string $code
+     *
+     * @return string
+     */
+    private function unindentCode($code)
+    {
+        $formattedCode = $code;
+        $codeLines = explode("\n", $code);
+
+        $indentedLines = array_filter($codeLines, function ($lineOfCode) {
+            return '' === $lineOfCode || '    ' === substr($lineOfCode, 0, 4);
+        });
+
+        if (count($indentedLines) === count($codeLines)) {
+            $formattedCode = array_map(function ($lineOfCode) { return substr($lineOfCode, 4); }, $codeLines);
+            $formattedCode = implode("\n", $formattedCode);
+        }
+
+        return $formattedCode;
+    }
+
+    // the name of the Twig extension must be unique in the application
+    public function getName()
```

Comment:
__[development][code_sniffer]__
* Declare public methods first,then protected ones and finally private ones
* You must use "/**" style comments for a function comment

<!-- Comment by [SideCI](https://sideci.com) -->